File: 7.gün/my_py_pkg(1)/my_py_pkg/my_first_node.py
#! /usr/bin/env python3
import rclpy
from rclpy.node import Node

class Mynode(Node):

    def __init__(self):
        super().__init__("py_test")
        self.counter_=0
        self.get_logger().info("merhaba ros2")
        self.create_timer(0.5 , self.timer_callback)

    def timer_callback(self):
        self.counter_ +=1
        self.get_logger().info("selam  "+str(self.counter_))

def main(args=None):
    rclpy.init(args=args)
    node = Mynode()
    rclpy.spin(node)
    rclpy.shutdown()

  

if __name__=="__main__":
    main()

File: 10.gün/my_cpp_pkg(8)/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"
#include <iostream>

class Mynode: public rclcpp::Node
{
public:
    Mynode(): Node("cpp_deneme"){
     RCLCPP_INFO(this-> get_logger(),"merhaba cpp");
     timer_=this->create_wall_timer(std::chrono::seconds(1),
                                    std::bind(&Mynode::timerCallback,this));
    } 

private:
  void timerCallback(){
    counter_++;
    RCLCPP_INFO(this-> get_logger(),"nasilsin %d", counter_);
  }
  rclcpp::TimerBase::SharedPtr timer_;
  int counter_;
};


int main(int argc,char **argv){
    rclcpp::init(argc,argv);
    auto node=std::make_shared<Mynode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}